diff --git a/dmenu.c b/dmenu.c
index cd556f4..9c915fc 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -60,6 +60,8 @@ static XIC xic;
 static Drw *drw;
 static Clr *scheme[SchemeLast];
 
+char layout[100];
+
 #include "config.h"
 
 static char * cistrstr(const char *s, const char *sub);
@@ -108,6 +110,10 @@ cleanup(void)
 	drw_free(drw);
 	XSync(dpy, False);
 	XCloseDisplay(dpy);
+
+  char switchbackC[114] = "xkb-switch -s ";
+  strcat(switchbackC, layout);
+  system(switchbackC);
 }
 
 static char *
@@ -711,6 +717,7 @@ static void
 run(void)
 {
 	XEvent ev;
+  system("xkb-switch -s us");
 
 	while (!XNextEvent(dpy, &ev)) {
 		if (XFilterEvent(&ev, win))
@@ -861,6 +868,9 @@ main(int argc, char *argv[])
 {
 	XWindowAttributes wa;
 	int i, fast = 0;
+  FILE *flayout = popen("xkb-switch", "r");
+  fgets(layout, 100, flayout);
+  pclose(flayout);
 
 	for (i = 1; i < argc; i++)
 		/* these options take no arguments */
