diff --git a/dmenu.c b/dmenu.c
index 1df3c9a..1b5c660 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -537,10 +537,10 @@ keypress(XKeyEvent *ev)
 			goto draw;
 		case XK_g: ksym = XK_Home;  break;
 		case XK_G: ksym = XK_End;   break;
-		case XK_h: ksym = XK_Up;    break;
-		case XK_j: ksym = XK_Next;  break;
-		case XK_k: ksym = XK_Prior; break;
-		case XK_l: ksym = XK_Down;  break;
+ 		case XK_h: ksym = XK_Prior;    break;
+ 		case XK_j: ksym = XK_Down;  break;
+ 		case XK_k: ksym = XK_Up; break;
+ 		case XK_l: ksym = XK_Next;  break;
 		default:
 			return;
 		}
